modules/search/src/main/java/com/sygic/maps/module/search/SearchFragment.kt

```diff
@@ -0,0 +1,158 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.module.search
+
+import android.content.Context
+import android.os.Bundle
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.CallSuper
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProviders
+import com.sygic.maps.module.common.delegate.ModulesComponentDelegate
+import com.sygic.maps.module.search.callback.SearchResultCallback
+import com.sygic.maps.module.search.databinding.LayoutSearchBinding
+import com.sygic.maps.module.search.di.DaggerSearchComponent
+import com.sygic.maps.module.search.viewmodel.SearchFragmentViewModel
+import com.sygic.maps.tools.viewmodel.factory.ViewModelFactory
+import com.sygic.maps.uikit.viewmodels.common.initialization.SdkInitializationManager
+import com.sygic.sdk.online.OnlineManager
+import com.sygic.sdk.position.GeoCoordinates
+import com.sygic.sdk.search.SearchResult
+import javax.inject.Inject
+
+const val SEARCH_FRAGMENT_TAG = "search_fragment_tag"
+
+/**
+ * A *[SearchFragment]* is the core component for any search operation. It can be easily used to display search input
+ * and search result list on the same screen. It can be modified with initial search input or coordinates. It comes with
+ * several pre build-in elements such as [SearchToolbar] or [SearchResultList]. TODO imports (MS-5212, MS-5213)
+ */
+@Suppress("unused", "MemberVisibilityCanBePrivate")
+class SearchFragment : Fragment(), SdkInitializationManager.Callback {
+
+    private val modulesComponent = ModulesComponentDelegate()
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelFactory
+    @Inject
+    internal lateinit var sdkInitializationManager: SdkInitializationManager
+
+    private lateinit var fragmentViewModel: SearchFragmentViewModel
+
+    private var injected = false
+    private fun inject() {
+        if (!injected) {
```

Comment:
sa nam to vola viackrat? alebo kvoli nejakym rotaciam?

Reply:
volá se to z onInflate() a z onAttach(), stejně jako v MapFragmentWrapper-u

Reply:
jo viem, len mi to pride take divne :/

Reply:
však my jsme to tam dali, pokud máš lepší nápad, sem s tím :-D